openpathsampling/tests/analysis/utils.py

```diff
@@ -0,0 +1,500 @@
+"""
+Tools for generating sequences of steps for analysis tests.
+
+The idea here is to have client code specify things like which snapshot
+index is the shooting point and what partial trajectory from one-way
+shooting should be, then we patch those things and use the internals of the
+mover to generate the real data.
+
+In this way, we use as much of the actual OPS machinery as possible,
+ensuring, for example, that the details we return are correct.
+
+Note that for the classes here, each instance represents a single move
+(single MC step). This is unlike the PathMover objects in OPS, which
+represents an individual type of move that can be reused for many steps.
+"""
+
+import numpy as np
+import random
+try:
+    from unittest.mock import Mock, patch
+except ImportError:
+    from mock import Mock, patch
+
+from openpathsampling.tests.test_helpers import make_1d_traj
+import openpathsampling as paths
+
+def _get_only(iterable, condition, error_msg):
+    possibilities = [item for item in iterable if condition(item)]
+    if len(possibilities) != 1:
+        msg = "expected 1 {error_msg}; found {nfound}: found".format(
+            error_msg=error_msg,
+            nfound=len(possibilities),
+            found=possibilities
+        )
+        raise AnalysisTestSetupError(msg)
+    return possibilities[0]
+
+
+class AnalysisTestSetupError(Exception):
+    """Raised for when an internal error occurs during test setup.
+
+    These usually indicate a problem with test suite, not with the code
+    itself.
+    """
+
+def make_trajectory(lower, upper):
+    """Make a trajectory from `lower` to `upper`.
+
+    This makes a trajectory segment with x-values of frames spread by 0.1.
+    For each trajectory, some random value epsilon is added, such that the
+    lowest value is ``lower + epsilon`` and the highest value is ``upper +
+    epsilon``.
+
+    Parameters
+    ----------
+    lower : float
+        the lower bound x-value of the trajectory
+    upper: float
+        the upper bound x-value of the trajectory (NOTE: trajectories will
+        cross this value, but not cross ``upper + 0.1``)
+
+
+    Returns
+    -------
+    :class:`.Trajectory`
+        the trajectory segment
+    """
+    xvals = np.arange(lower, upper + 0.01, 0.1) + np.random.random() * 0.1
+    return make_1d_traj(xvals)
+
+def make_tis_trajectory(cv_max, lower_bound=-0.1):
+    """Make a TIS trajectory with a given maximum x value.
+
+    """
+    increasing = make_trajectory(lower_bound, cv_max)
+    if cv_max >= 1.0:
+        return increasing
+    else:
+        decreasing = increasing.reversed[1:]
+        return increasing + decreasing
+
+def _select_by_input_ensembles(movers, ensembles):
```

Comment:
2 blank lines expected
```suggestion


def _select_by_input_ensembles(movers, ensembles):
```